mautic/mautic
# [TPROD-168] Remove OAuth1 support

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | no
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | yes
| Automated tests included?              | N/A
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/issues/202
| Issue(s) addressed                     | Fixes #... 




#### Description:

As discussed in https://mautic.atlassian.net/browse/TPROD-190, this PR removes OAuth1 support from Mautic due to the increasing workload to keep supporting it



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. In the top right corner of the screen, click the gear icon > API credentials
3. In every dropdown, you will now only have the OAuth2 option left

![image](https://user-images.githubusercontent.com/17739158/121944908-7b53c800-cd53-11eb-9202-8c5957bdb89a.png)

![image](https://user-images.githubusercontent.com/17739158/121944808-5d866300-cd53-11eb-8041-d5c80e72dea6.png)

Ensure OAuth2 is still working by doing the following:
1. Go to the API credentials screen mentioned above
2. Click "new"
3. Enter the following data (we won't use the Redirect URI thanks to https://github.com/mautic/mautic/pull/9837, so you can fill out anything there)

![image](https://user-images.githubusercontent.com/17739158/121945565-32e8da00-cd54-11eb-8ad1-ce69c1def755.png)

4. You will now see a public and secret key that you can use
5. Open Postman
6. Make sure you **exactly** copy the options below:

![image](https://user-images.githubusercontent.com/17739158/121946262-097c7e00-cd55-11eb-84d7-c0bdb16c5775.png)

7. After you got the token (which is the confirmation that OAuth2 still works), you can execute the API call with "Send"
